include/ur_client_library/comm/pipeline.h

```diff
@@ -349,62 +365,18 @@ class Pipeline
   moodycamel::BlockingReaderWriterQueue<std::unique_ptr<T>> queue_;
   std::atomic<bool> running_;
   std::thread pThread_, cThread_;
+  bool producer_fifo_scheduling_;
 
   void runProducer()
   {
     URCL_LOG_DEBUG("Starting up producer");
-    std::ifstream realtime_file("/sys/kernel/realtime", std::ios::in);
-    bool has_realtime;
-    realtime_file >> has_realtime;
-    if (has_realtime)
+    if (producer_fifo_scheduling_)
     {
+      pthread_t this_thread = pthread_self();
       const int max_thread_priority = sched_get_priority_max(SCHED_FIFO);
-      if (max_thread_priority != -1)
-      {
-        // We'll operate on the currently running thread.
-        pthread_t this_thread = pthread_self();
-
-        // struct sched_param is used to store the scheduling priority
-        struct sched_param params;
-
-        // We'll set the priority to the maximum.
-        params.sched_priority = max_thread_priority;
-
-        int ret = pthread_setschedparam(this_thread, SCHED_FIFO, &params);
-        if (ret != 0)
-        {
-          URCL_LOG_ERROR("Unsuccessful in setting producer thread realtime priority. Error code: %d", ret);
-        }
-        // Now verify the change in thread priority
-        int policy = 0;
-        ret = pthread_getschedparam(this_thread, &policy, &params);
-        if (ret != 0)
-        {
-          std::cout << "Couldn't retrieve real-time scheduling paramers" << std::endl;
-        }
-
-        // Check the correct policy was applied
-        if (policy != SCHED_FIFO)
-        {
-          URCL_LOG_ERROR("Producer thread: Scheduling is NOT SCHED_FIFO!");
-        }
-        else
-        {
-          URCL_LOG_INFO("Producer thread: SCHED_FIFO OK");
-        }
-
-        // Print thread scheduling priority
-        URCL_LOG_INFO("Thread priority is %d", params.sched_priority);
-      }
-      else
-      {
-        URCL_LOG_ERROR("Could not get maximum thread priority for producer thread");
-      }
-    }
-    else
-    {
-      URCL_LOG_WARN("No realtime capabilities found. Consider using a realtime system for better performance");
+      setFiFoScheduling(this_thread, max_thread_priority);
     }
+    std::ifstream realtime_file("/sys/kernel/realtime", std::ios::in);
```

Comment:
Does this do anything?

Reply:
No, this is a copy_paste error from copying the check where we actually checked whether the system does run a realtime kernel. Since we want to use FIFO scheduling independent of this, this line can be removed.

Reply:
Removed that line in the latest commit.